### PR TITLE
changed signup page from erb to haml format

### PR DESCRIPTION
**Pull request by sreeramgoalla**
sign_up

### CODE REVIEW

File: app/views/devise/registrations/new.html.haml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  .form-inputs
+    = f.input :given_name, label: "First Name", required: true, autofocus: true
+    = f.input :last_name, label: "Last Name", required: true
+    = f.input :brith_date, label: "Date of Birth", required: true
+    = f.input :email, label: "Email", required: true
+    = f.input :password, label: "Password", required: true, hint: ("#{@minimum_password_length} characters minimum" if @validatable)
+    = f.input :password_confirmation, label: "Confirm Password", required: true
+  .form-actions
+    = f.button :submit, "Sign up"
+    = render "devise/shared/links"
```
Comment (lemattma) [inline app/views/devise/registrations/new.html.haml]: This line should go two levels to the left. Check indentation.

**Comment by lemattma** [comment]
Good job @sreeramgoalla! Please include all the devise view files in this PR. Don't merge before we review.

File: app/views/devise/registrations/new.html.haml (hunk 1 above)
Comment (sreeramgoalla) [inline app/views/devise/registrations/new.html.haml]: which line number?

Comment (lemattma) [inline app/views/devise/registrations/new.html.haml]: @sreeramgoalla where the comment is. In Github you comment to lines.

Comment (lemattma) [inline app/views/devise/registrations/new.html.haml]: `= render "devise/shared/links"`

Comment (sreeramgoalla) [inline app/views/devise/registrations/new.html.haml]: @lemattma I've just updated all the device related erb's to haml files. Please let me know if it needs any corrections ?

File: app/views/devise/mailer/confirmation_instructions.html.haml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+%p Welcome #{@email}!
+%p You can confirm your account email through the link below:
+%p
```
Comment (lemattma) [inline app/views/devise/mailer/confirmation_instructions.html.haml]: `%p= link_to..` same line

File: app/views/devise/mailer/reset_password_instructions.html.haml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+%p Hello #{@resource.email}!
+%p Someone has requested a link to change your password. You can do this through the link below.
+%p
```
Comment (lemattma) [inline app/views/devise/mailer/reset_password_instructions.html.haml]: same line

File: app/views/devise/mailer/unlock_instructions.html.haml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+%p Hello #{@resource.email}!
+%p Your account has been locked due to an excessive number of unsuccessful sign in attempts.
+%p Click the link below to unlock your account:
+%p
```
Comment (lemattma) [inline app/views/devise/mailer/unlock_instructions.html.haml]: same line

File: app/views/devise/passwords/edit.html.haml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+%h2 Change your password
+= simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f|
+
+  = f.input :reset_password_token, as: :hidden
+  = f.full_error :reset_password_token
+
+  .form-inputs
+    = f.input :password, label: "New password", required: true, autofocus: true
```
Comment (lemattma) [inline app/views/devise/passwords/edit.html.haml]: Please use single quote unless you are interpolating `#{ ... }`

File: app/views/devise/passwords/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+%h2 Forgot your password?
+%br
```
Comment (lemattma) [inline app/views/devise/passwords/new.html.haml]: `<br>` is not intended for margin or spacing in general. Use CSS for this. I think @arbins created some css classes for this cases.

File: app/views/devise/registrations/edit.html.haml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+    = f.input :password, autocomplete: 'off', hint: "leave it blank if you don't want to change it", required: false
+    = f.input :password_confirmation, required: false
+    = f.input :current_password, hint: 'we need your current password to confirm your changes', required: true
+
+  .form-actions
+    = f.button :submit, 'Update'
+
+%h3 Cancel my account
+%p Are you Unhappy?
```
Comment (lemattma) [inline app/views/devise/registrations/edit.html.haml]: this will throw an error, if you have multiple lines, they should all be nested below `%p`

**Comment by lemattma** [comment]
Please rebase `master` and resolve the conflicts.

**Comment by lemattma** [comment]
:+1:

**Pull request merged**